ksql-engine/src/main/java/io/confluent/ksql/function/udf/map/ElementKudf.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.ksql.function.udf.map;
+
+import java.util.List;
+import java.util.Map;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+
+@UdfDescription(name = "element", author = "Confluent",
+    description = "Returns the specified element from a map or array.")
+public class ElementKudf {
+
+  @Udf(description = "Returns the value corresponding to the requested key in a map.")
+  public Object element(final Map<String, ?> inputMap, String key) {
+    if (inputMap == null) {
+      return null;
+    }
+    return inputMap.get(key);
+  }
+
+  @SuppressWarnings("rawtypes")
+  @Udf(description = "Returns the element at the specified index (counting from 0) in an array.")
```

Comment:
Isn't SQL convention normally to be base-1, rather than base-0?  Or has this changed with newer products?

Reply:
yes indeed - this was done just to align with the rest of ksql back at the time. seeing as we've subsequently updated other parts of the code to be 1-based, the same change should be applied here too